app/src/main/java/com/capstone/event_finder/activities/SignUpActivity.java

```diff
@@ -26,16 +36,76 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         Button btnSignUp = findViewById(R.id.btnSignUp);
 
+        setUpEventCategoryDropdown();
         btnSignUp.setOnClickListener(v -> signUpUser());
     }
 
+    private void setUpEventCategoryDropdown() {
+        boolean[] selectedCategories;
+        ArrayList<Integer> categoryList = new ArrayList<>();
+        String[] categoryArray = {"music", "visual-arts", "performing-arts", "film",
+                "lectures-books", "fashion", "food-and-drink", "festivals-fairs", "charities",
+                "sports-active-life", "nightlife", "kids-family"};
+
+        selectedCategories = new boolean[categoryArray.length];
+
+        tvEventCategoryDropdown.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                AlertDialog.Builder builder = new AlertDialog.Builder(SignUpActivity.this);
+                builder.setTitle("Select favorite event categories :");
+                builder.setCancelable(false);
+
+                builder.setMultiChoiceItems(categoryArray, selectedCategories, (dialog, i, isCategoryChecked) -> getCategoryListBasedOnSelection(i, isCategoryChecked, categoryList));
+
+                setUpSaveButton(builder);
+                builder.show();
+            }
+
+            private void setUpSaveButton(AlertDialog.Builder builder) {
+                builder.setPositiveButton("Save", (dialogInterface, i) -> {
+                    StringBuilder stringBuilder = new StringBuilder();
+                    categoryListOfStrings = new ArrayList<>();
+                    for (i = 0; i < categoryList.size(); i++) {
+                        String category = categoryArray[categoryList.get(i)];
+                        stringBuilder.append(category);
+                        categoryListOfStrings.add(category);
+                        if (i != categoryList.size() - 1) {
+                            stringBuilder.append(", ");
+                        }
+                    }
+                    allInterestCategories = stringBuilder.toString();
+                    tvEventCategoryDropdown.setText(allInterestCategories);
+                });
+            }
+        });
+    }
+
+    private void getCategoryListBasedOnSelection(int i, boolean isCategoryChecked, ArrayList<Integer> categoryList) {
+        if (isCategoryChecked) {
+            categoryList.add(i);
+            Collections.sort(categoryList);
+        } else {
+            categoryList.remove(Integer.valueOf(i));
+        }
+    }
+
     private void signUpUser() {
         ParseUser user = new ParseUser();
         user.setUsername(etUsername.getText().toString());
         user.setPassword(etPassword.getText().toString());
         user.put("zip", etZip.getText().toString());
+        user.put("bio", etBio.getText().toString());
+        if (categoryListOfStrings == null) {
+            Toast.makeText(SignUpActivity.this, "Error: Sign up failed", Toast.LENGTH_LONG).show();
```

Comment:
💡 Usually if a failure is due to the user, we want to clearly communicate what they can do to address the error (in this case, ensure they select at least one category).

Reply:
Will fix this!